### DISCUSSION TITLE
Grid system  row-span/col-span  JIT mode

### DISCUSSION THREAD

**Discussion by slaatarsi** [tailwindlabs/tailwindcss#4947] [Ideas]

**Comment by Martleby** [comment]
This would be very cool and ultra flexible. I hope it's possible because it would be a real game changer for using grid with Tailwind.

**Comment by mrtnpro** [comment]
Related question: https://github.com/tailwindlabs/tailwindcss/discussions/4381

**Comment by simonswiss** [comment]
Hey!

I have just [opened a pull request](https://github.com/tailwindlabs/tailwindcss/pull/5148) that will add support for this, once merged.

The utilities you mentioned are generated by the `gridColumn` plugin, and the class itself (the prefix) is only `col-`. The `span-` part is part of the _modifier_ for each value, so the arbitrary value will be a little different:

```grid-[span,17,/,span,17]```

Since arbitrary values [cannot contain spaces](https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support), you need to use commas here to obtain the correct syntax.

![image](https://user-images.githubusercontent.com/485747/128117040-444d4c26-2427-405f-9f9a-75ea185dea0d.png)

That syntax won't work just yet, but this PR brings support for it 👍

**Reply by Martleby** [reply]
That's great! Thanks very much Simon - I think this will be a hugely powerful addition to Tailwind!
